test/widgets/action_sheet_test.dart

```diff
@@ -1061,6 +1061,82 @@ void main() {
       });
     });
 
+    group('MarkTopicAsReadButton', () {
+      Future<void> setupToTopicActionSheetWithUnreadMessages(WidgetTester tester, {
```

Comment:
Looks like this and `setupToTopicActionSheetWithNoUnreadMessages` are each only called once in their respective test. Let's reorganize this for clarity.

The logic tested here whether `MarkTopicAsReadButton` is hidden in the action sheet, where the only variable is the presence of unread messages in the store, so it would make sense to have a single helper that's capable of controlling that.

If you look at the "topic action sheet" test group, you can find a helper called `prepare` that should accomplish all of these for you. You can use the "UserTopicUpdateButton" test group as a reference.

Reply:
The following comments apply to this helper, but you can skip most of them when working on the next revision.